src/shared/utils.ts

```diff
@@ -11,7 +11,8 @@ import * as open from 'open';
 export const getAliasByUsername = async (username: string): Promise<string> => {
   const alias = await Aliases.create(Aliases.getDefaultOptions());
   const keys = alias.getKeysByValue(username);
-  return keys?.length ? keys[0] : undefined;
+  // use the most recently added alias for that username
+  return keys?.length ? keys[keys.length - 1] : undefined;
```

Comment:
@mshanemc is alias list associated with a user guaranteed to remain in order aliases were created?

Reply:
1. That's how it seems to behave.
2. We're not doing any sorting of the object keys in core that I could see
https://github.com/forcedotcom/sfdx-core/blob/107070d43cfc223c7b4fb9bfa6c1ec1c10166b97/src/config/configGroup.ts#L217